### ISSUE TITLE
CellML support: allow for more readable unit names

### ISSUE THREAD

**Issue by agarny** [opencor/opencor#1158]
Right now, if we want to define the unit `Joules per Coulomb`, we would do the following using the `CellML Text` view:

```
def unit J_per_C as
    unit joule;
    unit coulomb {expo: -1};
enddef;
```

However, this is not great. It would be better if we could do the following instead:

```
def unit J/C as
    unit joule;
    unit coulomb {expo: -1};
enddef;
```

or as:

```
def unit J.C^-1 as
    unit joule;
    unit coulomb {expo: -1};
enddef;
```

Similarly, if we wanted to define `square meters per second`, we would have something like:

```
def unit m^2/s as
    unit meter {expo: 2};
    unit second {expo: -1};
enddef;
```

Then, it would be up to the `CellML Text` view (through our CellML support plugin) to convert those unit names into valid CellML identifiers, e.g. `J/C`, `J.C^-1` and `m^2/s` would be converted to `J_perC`, `J_times_C_superscript_minus_1` and `m_superscript_2_per_s`, respectively (and vice versa when opening a CellML file in the `CellML Text` view).

Those (more readable) unit names would also be used elsewhere in OpenCOR, e.g. in the `CellML Annotation` and `Single Cell` views.

**Comment by mirams** [comment]
Would you convert back as well? Otherwise save and close and reload presumably all still goes via CellML? I can imagine user confusion if OpenCOR shows something different to other CellML tools...

**Comment by agarny** [comment]
> Would you convert back as well?

Yes, see "... (and vice versa when opening a CellML file in the CellML Text view)" in my original message.

> Otherwise save and close and reload presumably all still goes via CellML?

Yes, to have something like `J/C` or `m^2/s` would only be for user convenience. When it comes to CellML, it would be references as `J_per_C` and `m_superscript_2_per_s`, respectively, since CellML doesn't allow characters such as `/` or `^`.

> I can imagine user confusion if OpenCOR shows something different to other CellML tools...

Internally, it would be the same, it's just the presentation of the information that would be more readable in, say, the `CellML Text` view.

Anyway, this comes from a user request and I think it's a reasonable one, don't you?

**Comment by nickerso** [comment]
One idea would be to keep the CellML text view as it is, but using the same math preview approach provide a "rendered" version of the units. Similarly, the Single Cell view could show the rendered units rather than the CellML unit names. This way when editing the CellML in either the CellML text or raw CellML view you would see the same thing (as in other tools also) yet there would be a convenient way to quickly see something more understandable to most users.

**Comment by agarny** [comment]
Thanks for the idea, I like it!
